src/set_relation/set_relation.cc

```diff
@@ -193,6 +193,32 @@ void Conjunction::reset() {
     mInequalities.clear();
 }
 
+//! Restrict this (interpreted as a Relation) to rhs, which is interpreted
```

Comment:
This notation that is in the comments multiple places and in the pull request description is incorrect.  r(s) is the notation for applying a relation to a set.  That is not what you are doing.  You are restricting the domain of a relation with a set.  I believe the omega paper used r restrict domain with s or something like that.

Also, C[x/z] means you are substituting the constraints on x with the constraints on z.  Is that really want you want to do?  Since the x vector is still in the tuple declaration don't you want to rewrite the D constraints substituting x for z?

Also make sure to indicate that the length of the x and z vectors (or arity of the input tuple and set) need to be equivalent.

Reply:
Yes you are right omega uses r \ s . 

For C[x/z] I got this notation from the paper this stems from the fact that x = z. In a restricted domain operation

```
   r = { x -> y : C }
   s = { z : D }
   r \ s = { x -> y : D && C && x = z  }
```

Reply:
I have fixed the notations.